tests/Game.class.test.js

```diff
@@ -0,0 +1,724 @@
+/* eslint-disable max-len */
+'use strict';
+
+const Game = require('../src/modules/Game.class');
+
+describe('Game class', () => {
+  it('should be a function', () => {
+    expect(typeof Game).toBe('function');
+  });
+
+  it('should have a constructor', () => {
+    expect(typeof Game.prototype.constructor).toBe('function');
+  });
+
+  describe('core methods', () => {
+    let game2048;
+
+    beforeEach(() => {
+      game2048 = new Game();
+    });
+
+    [
+      'getState',
+      'getScore',
+      'getStatus',
+      'moveLeft',
+      'moveRight',
+      'moveUp',
+      'moveDown',
+      'start',
+      'restart',
+    ].forEach((method) => {
+      it(`should have a "${method}" method`, () => {
+        expect(typeof game2048[method]).toBe('function');
+      });
+    });
+  });
+
+  describe('initial state', () => {
+    it('should have a state of an empty board by default', () => {
+      const game2048 = new Game();
+
+      const state = game2048.getState();
+
+      expect(state).toEqual([
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ]);
+    });
+
+    it('should have a score of 0 by default', () => {
+      const game2048 = new Game();
+
+      expect(game2048.getScore()).toBe(0);
+    });
+
+    it('should have a status of "idle" by default', () => {
+      const game2048 = new Game();
+
+      expect(game2048.getStatus()).toBe('idle');
+    });
+
+    it('should have a state of a custom board if the initial state is provided', () => {
+      const initialState = [
+        [2, 0, 0, 0],
+        [0, 4, 0, 0],
+        [0, 0, 8, 0],
+        [0, 0, 0, 16]];
+
+      const game2048 = new Game(initialState);
+
+      const state = game2048.getState();
+
+      expect(state).toEqual(initialState);
+    });
+
+    it('should have a score of a custom value if the initial score is provided', () => {
+      const initialScore = 42;
+
+      const game2048 = new Game(null, initialScore);
+
+      expect(game2048.getScore()).toBe(initialScore);
+    });
+
+    it('should have a status of a custom value if the initial status is provided', () => {
+      const initialStatus = 'win';
+
+      const game2048 = new Game(null, null, initialStatus);
+
+      expect(game2048.getStatus()).toBe(initialStatus);
+    });
+  });
+
+  describe('after start', () => {
+    it('should have a score of 0 after start', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      expect(game2048.getScore()).toBe(0);
+    });
+
+    it('should have a status of "playing" after start', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should have a state with two random cells with 2 or 4 after start', () => {
```

Comment:
the original game generate 4 in 20% of cases otherwise 2
So i would just test that there are 2 values <= 4